lib/source/pl/core/evaluator.cpp

```diff
@@ -1006,6 +1007,18 @@ namespace pl::core {
         if (this->isDebugModeEnabled())
             this->m_console.log(LogConsole::Level::Debug, fmt::format("Base Pattern size: 0x{:02X} bytes", sizeof(ptrn::Pattern)));
 
+        this->m_sourceLineLength.clear();
+        for (auto &topLevelNode : ast) {
+            if (topLevelNode->getLocation().source->source == "<Source Code>") {
```

Comment:
Instead of checking for `<Source Code>` manually here (which does not always mean it's the top level file), instead use the `source->mainSource` member

Reply:
OK, change was made as requested change. It seems like there may be edge cases when <Source Code> is not what is in the pattern editor because from what I can tell it should be but I take your word that mainSource will get it 100% of the time.